tests/kw_env_test.sh

```diff
@@ -195,6 +198,40 @@ function test_use_target_env_invalid_env()
   assertEquals "($LINENO) Env does not exists" "$?" 22
 }
 
+function test_validate_env_before_use_invalid_case_with_config()
+{
+  local output
+
+  output=$(validate_env_before_use)
+  assertEquals "(${LINENO}) it should not allow switch to a new env due to .config file." "$?" 22
+}
+
+function test_validate_env_before_use_invalid_case_with_object_file()
+{
+  local output
+
+  # Remove the .cofig file
+  mv '.config' 'config'
+
+  # Create a fake object file
+  mkdir -p 'drivers/gpu/drm/'
+  touch 'drivers/gpu/drm/drm_atomic.o'
+
+  output=$(validate_env_before_use)
+  assertEquals "(${LINENO}) it should not allow switch to a new env due to object files." "$?" 22
+}
+
+function test_validate_env_before_use_no_config_and_no_object_files()
+{
+  local output
+
+  # Remove the .cofig file
+  mv '.config' 'config'
+
+  output=$(validate_env_before_use)
+  assertEquals "(${LINENO}) it should not allow switch to a new env due to object files." "$?" 0
```

Comment:
Also, the message 'it should not allow switch to a new env due to object files.' seems wrong. Shouldn't it be something like '**it should** allow switch to a new env.'?